runner/internal/shim/task.go

```diff
@@ -4,17 +4,19 @@ import (
 	"context"
 	"crypto/sha256"
 	"fmt"
+	"log"
 	"sync"
 )
 
 type TaskStatus string
 
 const (
-	// pending -> pulling -> creating -> running -> terminated
-	//    |         |           |
-	//    v         v           v
-	// terminated terminated terminated
+	// pending -> preparing -> pulling -> creating -> running -> terminated
```

Comment:
@un-def, could you elaborate why preparing status is needed?

Reply:
The idea was to split `Run()` method into two — submission and run operations, `Submit()`, which should work as fast as possible, only puts a task into storage with `pending` status and returns an error on conflict, which is returned to the server via HTTP API. `Run()`, in turn, performs some preparation steps before it can start a container (puts authorized_keys, prepares volumes, etc.). Previously, these steps were executed with `pending` status (and with legacy API — `pulling` state), which is not quite right semantically.

In short, `penging` has not `Run()` yet, `preparing` — just first status in the `Run()` sequence (prepare, pull image, create container, run, mark terminated)